### ISSUE TITLE
ScaleLine causes error with debug build

### ISSUE THREAD

**Issue by beck21** [openlayers/openlayers#3760]
When using the ScaleLine control with the debug build the following error gets thrown:

ol-debug.js:4290 Uncaught AssertionError: Assertion failed: transform should be defined
goog.asserts.DEFAULT_ERROR_HANDLER @ ol-debug.js:4290
goog.asserts.doAssertFailure_ @ ol-debug.js:4325
goog.asserts.assert @ ol-debug.js:4354
ol.proj.getTransformFromProjections @ ol-debug.js:18803
ol.proj.Projection.getPointResolution_ @ ol-debug.js:18423
ol.proj.Projection.getPointResolution @ ol-debug.js:18462
ol.control.ScaleLine.updateElement_ @ ol-debug.js:84411
ol.control.ScaleLine.render @ ol-debug.js:84371
goog.events.EventTarget.fireListeners @ ol-debug.js:12811
goog.events.EventTarget.dispatchEventInternal_ @ ol-debug.js:12908
goog.events.EventTarget.dispatchEvent @ ol-debug.js:12723
ol.Map.renderFrame_ @ ol-debug.js:82875
goog.async.AnimationDelay.doAction_ @ ol-debug.js:36185

Removing ScaleLine from the controls object causes the error to go away. 
controls = ol.control.defaults().extend([
                    new ol.control.FullScreen(),
                    new ol.control.ZoomSlider(),
                    new ol.control.Rotate(),
                    new ol.control.OverviewMap(),
                    new ol.control.ScaleLine()
                ]);

The error does not exist when using the ol.js build, just ol-debug.js.

**Comment by alvinlindstam** [comment]
What projection are you using on the view? Are you using proj4js?

**Comment by beck21** [comment]
It is a Zoomify projection.

projection = new ol.proj.Projection({
        code: 'ZOOMIFY',
        units: 'pixels',
        extent: [0, 0, imgWidth, imgHeight],
        //getPointResolution: getPointResolution
    });

**Comment by alvinlindstam** [comment]
Well, that explains it. The ScaleLine control displays a "rough x-axis distances, calculated for the center of the viewport". Since this may vary depending on projection, it asks the projection for the resolution of the center coordinate.

The default implementation for getPointResolution calculates the resolution by transforming them to a global coordinate system and calculates their real distance on that. Since your projection is not geo-referenced, this won't work so you must provide a way to transform the pixel coordinates to real distances.

Take a look at http://jsfiddle.net/c8tpbpf2/. It's based on the zoomify example (http://openlayers.org/en/v3.6.0/examples/zoomify.html?q=zoomify), I just added the ScaleLine control and a simple getPointResolution function. I also had to change the projection unit to meters to handle a second assertion error (ScaleLine control expects the projection to have a ol.proj.ProjectionUnit as unit)
